lib/cache/cache.go

```diff
@@ -1408,7 +1416,13 @@ func (c *Cache) fetchAndWatch(ctx context.Context, retry retryutils.Retry, timer
 // cannot run concurrently with event processing.
 func (c *Cache) performRelativeNodeExpiry(ctx context.Context) error {
 	// TODO(fspmarshall): Start using dynamic value once it is implemented.
-	gracePeriod := apidefaults.ServerAnnounceTTL
+
+	// because event streams are not necessarily ordered across keys expiring on the
+	// server announce TTL may sometimes generate false positives. Using the watcher
+	// creation grace period as our safety buffer is mostly an arbitrary choice, but
+	// since it approximates our expected worst-case staleness of the event stream its
+	// a fairly reasonable one.
+	gracePeriod := apidefaults.ServerAnnounceTTL + backend.DefaultCreationGracePeriod
```

Comment:
This is using the default value for `DefaultCreationGracePeriod` (and for `ServerAnnounceTTL`, actually) because there's no way to specify a different value for it from outside the cache, right?

Reply:
Yes, it's not currently configurable.  Theres a long-standing TODO in this function to make it dynamically proportional to server TTL if we ever implement dynamic server TTL, but fundamentally I'm not sure theres much use is making this based on arbitrary input rather than being based on server TTL.

I suppose we could make the additional grace period configurable so that we could tune it without code changes.  I'm not sure how likely we are to hit that need though.  The specific issue we're running into here is that expiry was racing with refresh.  In reality changes are we could have fixed it with a much shorter grace period.  The ~3m grace period provided by `DefaultCreationGracePeriod` just seemed like a nice way to err on the side of caution.